### ISSUE TITLE
Error with lines that open with `{` and space.

### ISSUE THREAD

**Issue by howardroark** [laktak/hjson-py#3]
I am getting an error over this line ... https://github.com/stackstrap/environment/blob/master/environment/init.sls#L14

```
HjsonDecodeError: Key names that include {}[],: or whitespace require quotes at: line 14 column 5 (char 218)
```

To me this should work...  It is probable that the error roots with the whitespace between the `{` and the `t`, but that is not very intuitive if it is the case.  Could this be resolved?

Thanks!

**Comment by laktak** [comment]
Is the first line removed before it goes to Hjson? The error is actually reported on line 15.
```
  file.symlink: [
    { target: /usr/bin/ack-grep }
  ]
```
A quoteless string ends at the newline, so `}` is part of the string and Hjson thinks you are trying to create a key named `]`.

Use either `{ target: "/usr/bin/ack-grep" }` or
```
  { 
    target: /usr/bin/ack-grep 
  }
```

I'll try to come up with a better error message though.

**Comment by howardroark** [comment]
Ahhh yes you are right, the first line does get removed.  I should know that since the renderer I added to Salt does that :)

Just so I understand this, would the below also work?

```
  {  target: /usr/bin/ack-grep 
  }
```

Is it a technical infeasibility to have quoteless values end at the presence of the next character which would close the last opened type of object?  I'm not very good a following rules, like most humans ;)

**Comment by laktak** [comment]
Yes, your sample works as well.

> Is it a technical infeasibility to have quoteless values end at the presence of the next character which would close the last opened type of object?

Of course you could :) but that would be a breaking change and it would also complicate the rules for Hjson. I think having a single value in the object and wanting to write that on a single line is the exception rather than the norm.

**Comment by howardroark** [comment]
Well if it is a breaking change that is certainly no good!  To be honest I don't think well on those terms :P My brain is not very computer like.

I just look at it like this... Is it instinctual for people who write JSON to try it? If "yes".. attempt to cater for it.  More or less I would explain Hjson as... "a common sense way to write JSON with comments and without quotes and commas".  It would be very cool if humans could run with that simple "truth" and write Hjson without reading any rules.  If you try something that "feels right" but does not work... it is a subconscious detractor.

I'm just think how to get the "masses" using this :)

**Comment by laktak** [comment]
I understand what you mean but still, allowing this would make the rules harder to explain and not easier.

*A quoteless string ends at the new-line* is much simpler to explain.

**Comment by howardroark** [comment]
Well I certainly don't want to get annoying here! This project is in a great place :) 

I think in my head... *A quoteless string will always end at the new-line* ... (*but if you want you can also wrap it up by closing an object on the same line*)

**Comment by rainhard** [comment]
@howardroark +1, i think so too.

If special chars in string, you need wrap it up. And that will be more useful.
Is it possible to change it？

**Comment by laktak** [comment]
> If special chars in string, you need wrap it up. And that will be more useful.
> Is it possible to change it？

I'm not sure what you mean. If it's about the Hjson syntax please take a look at the issues in https://github.com/laktak/hjson/issues

I'm going to close this since the unhelpful error messages were fixed with f80f24d3a3f272dcd397b7129464642225521817.
